org.eclipse.scanning.api/src/org/eclipse/scanning/api/malcolm/attributes/MalcolmAttribute.java

```diff
@@ -49,11 +64,63 @@ public boolean isWriteable() {
 	public void setWriteable(boolean writeable) {
 		this.writeable = writeable;
 	}
+	/* (non-Javadoc)
+	 * @see org.eclipse.scanning.api.malcolm.attributes.IDeviceAttribute#getLabel()
+	 */
+	@Override
 	public String getLabel() {
 		return label;
 	}
 	public void setLabel(String label) {
 		this.label = label;
 	}
-	public abstract Object getValue();
+	/* (non-Javadoc)
+	 * @see org.eclipse.scanning.api.malcolm.attributes.IDeviceAttribute#getValue()
+	 */
+	@Override
+	public abstract T getValue();
+	
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((description == null) ? 0 : description.hashCode());
+		result = prime * result + ((label == null) ? 0 : label.hashCode());
+		result = prime * result + ((name == null) ? 0 : name.hashCode());
+		result = prime * result + Arrays.hashCode(tags);
+		result = prime * result + (writeable ? 1231 : 1237);
+		return result;
+	}
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)